src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -19,25 +22,50 @@
 
 public class Shooter extends SubsystemBase {
 
-    //CANSparkMax for the hopper
-    CANSparkMax m_chuteMotor;
-    //WPI_TalonFX for the shooter
-    WPI_TalonFX m_topShooterMotor, m_bottomShooterMotor;
-    //Limit Switch for Intake
-    DigitalInput m_limitSwitch1, m_limitSwitch2;
-
+    CANSparkMax m_chuteMotor; //CANSparkMax for the hopper
+    WPI_TalonFX m_topShooterMotor, m_bottomShooterMotor; //WPI_TalonFX for the shooter
+    DigitalInput m_limitSwitch1, m_limitSwitch2; //Limit Swit
     
+    static TreeMap<Double, ShooterSpeeds> shooterSpeeds = new TreeMap<>(Map.ofEntries(
+        Map.entry(1., new ShooterSpeeds(0.5, 0.5, 0.5)),
+        Map.entry(2., new ShooterSpeeds(0.5, 0.5, 0.5)),
+        Map.entry(3., new ShooterSpeeds(0.5, 0.5, 0.5))
+    ));
     //Shooter class constructor, initialize arrays for motors controllers, encoders, and SmartDashboard data
     public Shooter() {
-
         m_chuteMotor = new CANSparkMax(Constants.CHUTE_CAN_ID, MotorType.kBrushless);
 
         m_topShooterMotor = new WPI_TalonFX(Constants.TOP_SHOOTER_CAN_ID);
         m_bottomShooterMotor = new WPI_TalonFX(Constants.BOTTOM_SHOOTER_CAN_ID);
 
         m_limitSwitch1 = new DigitalInput(Constants.LIMIT_SWITCH_ONE);
         m_limitSwitch2 = new DigitalInput(Constants.LIMIT_SWITCH_TWO);
+    }
 
+    public static class ShooterSpeeds {
+        double top, bottom, chute;
+        public ShooterSpeeds(double top, double bottom, double chute) {
+            this.top = top;
+            this.bottom = bottom;
+            this.chute = chute;
+        }
+        public ShooterSpeeds interpolate(ShooterSpeeds other, double ratio) {
+            return new ShooterSpeeds(
+                top * (1 - ratio) + other.top * ratio,
```

Comment:
A cleaner and _very slightly_ faster expression would be:
   top + (other.top - top) * ratio
To me, this is much more understandable.